src/todo-config.ts

```diff
@@ -39,7 +39,21 @@ export function getTodoConfig(
 ): TodoConfig | undefined {
   const daysToDecayPackageConfig = getFromPackageJson(baseDir);
   const daysToDecayEnvVars = getFromEnvVars();
-  const mergedConfig = Object.assign({}, daysToDecayPackageConfig, daysToDecayEnvVars, todoConfig);
+  let mergedConfig = Object.assign({}, daysToDecayPackageConfig, daysToDecayEnvVars, todoConfig);
+
+  // we set a default config if the mergedConfig is an empty object, meaning either or both warn and error aren't
+  // defined and the package.json doesn't explicitly define an empty config (they're opting out of defining a todoConfig)
+  if (
+    Object.keys(mergedConfig).length === 0 &&
+    (typeof daysToDecayPackageConfig === 'undefined' ||
+      (typeof daysToDecayPackageConfig === 'object' &&
+        Object.keys(daysToDecayPackageConfig).length !== 0))
```

Comment:
I don't fully understand this specfic part of the conditional. If you remove these two lines, doesn't it work the same?

Reply:
It doesn't, because we need to explicitly handle the following cases:

- The fall-through of empty <- package.json <- env vars <- command line produces an empty object
- The package.json has no definition at all
- The package.json has explicitly an empty object

I think the above is satisfied only by that conditional, since using the Object.assign above will always default to `{}`, we need to disambiguate that from the package.json explicitly setting `{}`. If we just use the first part of the condition, we'd never be able to explicitly tell if they set the config to `{}` in package.json, as again, we're defaulting to `{}` in the fall-through.